src/Device/Driver/AR62xx.cpp

```
@@ -0,0 +1,573 @@
+
+  /** convert the passive-frequency to an AR62xx-ID */
+  uint16_t passive_frequency_idx = FrequencyToAR62xxId(passive_frequency);
+
+  /** command-start */
+  command[command_length++] = 0xA5;
+
+  /** add prot-ID */
+  command[command_length++] = 0x14;
+  command[command_length++] = 5;
+
+  /** set active station */
+  if (active_passive == ACTIVE_STATION)
+    active_frequency_idx = FrequencyToAR62xxId(frequency);
+
+  /** set passive station */
+  else
+    passive_frequency_idx = FrequencyToAR62xxId(frequency);
+
+  /** setting frequencies */
+  command[command_length++] = 22;
+  command[command_length++] = GetNthByte(active_frequency_idx, 1);
+  command[command_length++] = GetNthByte(active_frequency_idx, 0);
+  command[command_length++] = GetNthByte(passive_frequency_idx, 1);
+  command[command_length++] = GetNthByte(passive_frequency_idx, 0);
+
+  /** Creating the binary value */
+  uint16_t command_crc = CRCBitwise(command, command_length);
+  command[command_length++] = GetNthByte(command_crc, 1);
+  command[command_length++] = GetNthByte(command_crc, 0);
+  return command_length;
+}
+
+/**
+   * @brief parses the received data
+   * this is needed to save the active and passive station
+   * 
+   * @param string recived data
+   * @param len length of data
+   * @return true if data could be parsed
+   * @return false if data could not be parsed
+   */
+bool AR62xxDevice::ParseReceivedData(const char *string, size_t len)
+{
+  /** check if there is data */
+  if (string == NULL || len == 0)
+    return false;
+
+  size_t idx = 0;
+  uint16_t crc = 0;
+  static uint16_t rec_buf_len = 0;
+  int command_length = 0;
+#define REC_BUFSIZE 127
+  static uint8_t command[REC_BUFSIZE];
+
+  /** loop as long we have data */
+  while (idx < len)
+  {
+    /** check if we received a header*/
+    if ((uint8_t)string[idx] == 0xA5)
+      rec_buf_len = 0;
+
+    /** check if the buffer-size is reached */
+    if (rec_buf_len >= REC_BUFSIZE)
+      rec_buf_len = 0;
```

So.. this silently truncates the buffer... is this correct? Is that your concept, is that how the protocol is supposed to work?

---

yes

---

I have no idea what "yes" means. I asked two questions.

---

please have a look at the new implementation which is completely different from this and fully documented :-)